app/src/main/java/org/mozilla/fenix/settings/wallpaper/WallpaperSettings.kt

```diff
@@ -152,33 +256,44 @@ private fun WallpaperSnackbar(
 @Composable
 @Suppress("LongParameterList")
 private fun WallpaperThumbnails(
-    wallpapers: List<Wallpaper>,
+    onLearnMoreClick: () -> Unit,
+    wallpapers: Map<WallpaperProviderGroup, List<Wallpaper>>,
     defaultWallpaper: Wallpaper,
     loadWallpaperResource: suspend (Wallpaper) -> Bitmap?,
     selectedWallpaper: Wallpaper,
     numColumns: Int = 3,
     onSelectWallpaper: (Wallpaper) -> Unit,
 ) {
-    Column(modifier = Modifier.padding(vertical = 30.dp, horizontal = 20.dp)) {
-        val numRows = (wallpapers.size + numColumns - 1) / numColumns
-        for (rowIndex in 0 until numRows) {
-            Row {
-                for (columnIndex in 0 until numColumns) {
-                    val itemIndex = rowIndex * numColumns + columnIndex
-                    if (itemIndex < wallpapers.size) {
-                        Box(
-                            modifier = Modifier.weight(1f, fill = true).padding(4.dp),
-                        ) {
-                            WallpaperThumbnailItem(
-                                wallpaper = wallpapers[itemIndex],
-                                defaultWallpaper = defaultWallpaper,
-                                loadWallpaperResource = loadWallpaperResource,
-                                isSelected = selectedWallpaper == wallpapers[itemIndex],
-                                onSelect = onSelectWallpaper,
-                            )
+    wallpapers.forEach { (group, wallpapers) ->
+        if (wallpapers.isNotEmpty()) {
+            WallpaperGroups(
+                title = group.title,
+                onLearnMoreClick = onLearnMoreClick,
+            )
+        }
```

Comment:
I think we should move this bit up a level to `WallpaperSettings`. This will make it easier to re-use `WallpaperThumbnails`, as @mavduevskiy needs to in https://github.com/mozilla-mobile/fenix/pull/26758